src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -18029,6 +18029,28 @@ static object F((I<object>, I<object?>)? x, (I<object?>, I<object>) y)
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y").WithArguments("(I<object?>, I<object>)", "(I<object>, I<object?>)").WithLocation(10, 21));
         }
 
+        [Fact]
+        public void NullCoalescingOperator_RightIsUnreachable()
+        {
+            var source = @"
+class C
+{
+    object? F = null;
+    void M(C? c1, C c2)
```

Comment:
>C c2 [](start = 18, length = 4)

Consider using `C? c2` instead, to indicate whether the assignment `c1 = c2` is ignored.

Reply:
Never mind. As you explained offline, if the assignment was not ignored, `c1.F` would be considered nullable later.

---
In reply to: [246097028](https://github.com/dotnet/roslyn/pull/32225#discussion_r246097028) [](ancestors = 246097028)